libnavigation-core/src/test/java/com/mapbox/navigation/core/MapboxNavigationTest.kt

```diff
@@ -1486,7 +1486,7 @@ internal class MapboxNavigationTest : MapboxNavigationBaseTest() {
         }
 
     @Test
-    fun `set route - new routes immediately interrupts reroute`() = coroutineRule.runBlockingTest {
+    fun `set route - new routes interrupts reroute`() = coroutineRule.runBlockingTest {
```

Comment:
This should remain true. Reroute should be interrupted immediately without having to wait to be dispatched. Same below.

Reply:
Haha, that's funny because for some cases we can only know the reason under mutex... And reroute interruption is based on reason. But OK, I'll try to think of something.

Reply:
Actually, no, that's not possible.
We can only distinguish reason ALTERNATIVE and NEW inside the mutex.

Reply:
I have an alternative proposal: https://github.com/mapbox/mapbox-navigation-android/pull/6942